### PR TITLE
Added heartbeat fields add updated Makefile to be able to generate CRDs

### PR DESCRIPTION
**Pull request by OlegSternbergIBM**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,3 +6,8 @@
 [![License](https://img.shields.io/github/license/open-cluster-management/hub-of-hubs-data-types)](/LICENSE)
 
 The data-types module of [Hub-of-Hubs](https://github.com/open-cluster-management/hub-of-hubs).
+
+## DeepCopy and CRD generation
+
+1. DeepCopy - run ```make generate```. The file will be generated in the corresponding API folder
+2. CRD - run ```make manifests```. The YAML files will be created under <root>/crd folder
```
Comment (vadimeisenbergibm) [inline README.md:13]: Could you please add a sentence that after generating the CRDs, they must be copied into the APIs repo?

File: Makefile [hunk 2]
```diff
@@ -31,4 +31,34 @@ lint: clean-vendor
...
+	mv ./apis/config/go_temp.mod ./apis/config/go.mod
+	mv ./go_temp.mod ./go.mod
+
+generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	mv ./apis/config/go.mod ./apis/config/go_temp.mod
+	cp ./go.mod ./go_temp.mod
+	go mod tidy
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+	mv ./apis/config/go_temp.mod ./apis/config/go.mod
+	mv ./go_temp.mod ./go.mod
```
Comment (vadimeisenbergibm) [inline Makefile:64]: a newline missing

File: README.md (hunk 1 above)
Comment (OlegSternbergIBM) [reply README.md]: done

File: Makefile (hunk 2 above)
Comment (OlegSternbergIBM) [reply Makefile]: done

**Review by vadimeisenbergibm — APPROVED** [review]

**Pull request merged**